src/main/scala/com/chaordicsystems/cactus/Parser.scala

```diff
@@ -0,0 +1,140 @@
+package com.chaordicsystems.cactus
+
+import com.chaordicsystems.cactus.Operation._
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s._
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+
+case class InvalidCactusQueryFormatException(m: String = "There's a problem with your query, please check the documentation.") extends Exception
+case class Cactus(op: String, field: Option[String], args: Any)
+
+object Parser {
+  def mapToCactus(values: Map[String, Any]): Cactus = {
+    val op = values("op").toString
+    val field = if (values.contains("field")) Option(values("field").toString) else None
+    val args = values("args")
+    Cactus(op, field, args)
+  }
+
+  def comparativeQueryHandler(op: Operation, property: String, value: Any): QueryDefinition = {
+    op match {
+      case LT => rangeQuery(property) to value includeUpper false
+      case GT => rangeQuery(property) from value includeLower false
+      case LE => rangeQuery(property) to value includeUpper true
+      case GE => rangeQuery(property) from value includeUpper true
+      case _  => matchQuery(property, value)
+    }
+  }
+
+  def comparativeQuery(op: Operation, property: String, value: Any): QueryDefinition = {
+    value match {
+      case _: Boolean => termQuery(property, value)
+      case _: Double  => comparativeQueryHandler(op, property, value.asInstanceOf[Double])
+      case _: BigInt  => comparativeQueryHandler(op, property, value.asInstanceOf[BigInt].toInt)
+      case _: String  => comparativeQueryHandler(op, property, value.asInstanceOf[String])
+      case _          => throw InvalidCactusQueryFormatException()
+    }
+  }
+
+  def handleDetails(cactus: Cactus): NestedQueryDefinition = {
+    val name = cactus.field.get.split("\\.")(1)
+    val valueType = cactus.args match {
+      case _: BigInt  => "value_int"
+      case _: Boolean => "value_bool"
+      case _: Double  => "value_float"
+      case _: String  => "value_str"
+      case _          => throw InvalidCactusQueryFormatException()
+    }
+
+    Operation.withName(cactus.op) match {
+      case NE => nestedQuery("details").query {
+        bool {
+          must(matchQuery("details.name", name))
+          not(comparativeQuery(NE, s"details.$valueType", cactus.args))
+        }
+      }
+      case op => nestedQuery("details").query {
+        bool {
+          must (
+            matchQuery("details.name", name),
+            comparativeQuery(op, s"details.$valueType", cactus.args)
+          )
+        }
+      }
+    }
+  }
+
+  def handleCatOrTag(cactus: Cactus): QueryDefinition = {
+    val field = if (cactus.field.isDefined) cactus.field.get else throw InvalidCactusQueryFormatException()
+    val listArgs = cactus.args.asInstanceOf[List[Any]]
+
+    Operation.withName(cactus.op) match {
+      case ALL =>
+        if (listArgs.length == 1) {
```

Comment:
It's strange validating this kind of thing here. I'd move/encapsulate operators intelligence inside each operator type.